### PR TITLE
[MODULAR] Reagent weapon twohandedness, plus some QoL

### PR DESCRIPTION
**Pull request by burgerenergy**
## About The Pull Request

This does three individual things;

Firstly, it adds the twohanded module to most of the reagent forge weapons. The rough idea that if you're two handing a weapon, you get a x1.4 damage modifier. (I would add also block chance, but the bokken is kind of messy and it's not inherent to the twohanded datum.) This is a fairly significant steroid, but the way TG implemented wielding doesn't let you break it simply by switching your hands, so you're pretty stuck in and can't easily switch to shoving. (And, ofc, you can't bring a shield with you.) The hammer is the one stand out; it hits like a truck as originally, but does not _require_ you to twohand it to so much as carry it around, just to actually use it for combat.

If I had the ability, I would make the spear's 2 range require twohanding as TGMC has proved two tile range is one of the single most overpowered things in SS13, melee but guaranteed is hell. Alas!

Secondly, some breadcrumb descriptions have been added to forge items here and there, so a player hopefully doesn't have to glance at the guide to get through it.

Lastly, some QoL. The bokken's damage was reversed, so that's fixed. The dagger and the hand hatchet have been given tool status, as well as the ability to butcher corpses.

~~... In theory, the reason why this is drafted is because some snippet of the code there has it take _forever_ despite it having a time of eight seconds. This is also true for tool_knife use, like working food, so I assume I am missing some value somewhere. I'll have to dig into this.~~ fucking toolspeed

## How This Contributes To The Nova Sector Roleplay Experience

Let's make swords more spicy, without breaking the 30 force damage button that's the floor for powerful, if still one handed, antagonist weapons. A gun will still out DPS a sword realistically, though.

## Proof of Testing

image

## Changelog

:cl:
add: The larger Reagent Forge weapons can be wielded in two hands by using them in hand, increasing their damage significantly! The smaller Reagent Forge weapons, the dagger and axe, now can be used as tools, particularly a knife.
/:cl:

### CODE REVIEW

**Comment by Knouli** [comment]
Staff really really really shouldn't have \any\ force assigned to it. Regent imbuing is something tribals can still do, and staves fulfill the role of allowing for the application of beneficial regents without stabbing your 'friend.' Giving it any damage, even as weak as it is, removes this niche without giving it any other - being too weak to be considered an offensive weapon, but too damaging to consider friendly application.

**Comment by honkpocket** [comment]
I dread that you made this PR after my own I tried really hard to work together with contributors on, and you're just undoing work from that PR.

File: modular_nova/modules/reagent_forging/code/forge_weapons.dm [hunk 1]
```diff
@@ -280,7 +314,7 @@
 /obj/item/forging/reagent_weapon/bokken
 	name = "bokken"
 	desc = "A bokken that is capable of blocking attacks when wielding in two hands, possibly including bullets should the user be brave enough."
-	force = 20
+	force = 10
```
Comment (honkpocket) [inline modular_nova/modules/reagent_forging/code/forge_weapons.dm:317]: **NO!**

File: modular_nova/modules/reagent_forging/code/forge_weapons.dm [hunk 2]
```diff
@@ -167,7 +201,7 @@
 
 /obj/item/forging/reagent_weapon/hammer/Initialize(mapload)
 	. = ..()
-	AddComponent(/datum/component/two_handed, force_unwielded = 24, force_wielded = 24, require_twohands = TRUE)
```
Comment (honkpocket) [inline modular_nova/modules/reagent_forging/code/forge_weapons.dm:170]: I spent a lot of time working out how to balance the hammer in https://github.com/NovaSector/NovaSector/pull/404, and here you are undoing it. Why??? Again, where was your feedback when I was working hard on these items? Please don't change it again?

File: modular_nova/modules/reagent_forging/code/forge_weapons.dm [hunk 3]
```diff
@@ -27,7 +28,7 @@
 	belt_icon_state = "sword_belt"
 	hitsound = 'sound/weapons/bladeslice.ogg'
 	throwforce = 10
-	block_chance = 25
+	block_chance = 30
```
Comment (honkpocket) [inline modular_nova/modules/reagent_forging/code/forge_weapons.dm:31]: Maintainers concluded in https://github.com/NovaSector/NovaSector/pull/404 that this blockchance was fine, why add +5? I tried really hard to collect feedback during that testmerge. I have to admit seeing it get changed again a month later is frustrating, why didn't you suggest this during that month of time?

File: modular_nova/modules/reagent_forging/code/forge_weapons.dm [hunk 4]
```diff
@@ -56,10 +65,18 @@
 	attack_verb_simple = list("attack", "slash", "stab", "slice", "tear", "lacerate", "rip", "dice", "cut")
 	sharpness = SHARP_EDGED
 
+/obj/item/forging/reagent_weapon/katana/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/two_handed, force_unwielded = 20, force_wielded = 28)
```
Comment (honkpocket) [inline modular_nova/modules/reagent_forging/code/forge_weapons.dm:70]: Why two handed????? why does it need this?????

**Comment by Lutowski** [comment]
Hi, we just had changes applied to forging recently that this PR pretty much uproots. I'm not sure if this should've been test merged so quickly when honk's previous PR didn't even get to cook for a month.

Unless there's a glaring issue with the changes that we don't see, I'd really recommend just adding onto the existing changes that doesn't remove honk's contributions wholesale.

**Comment by Ephemeralis** [comment]
My understanding of things was force *20* was the floor for what was considered powerful weapon-wise. Force 30 is entering e-sword territory which is like, dedicated overpowered (by design) antagonist level stuff, isn't it? 

A force 28 katana that can be made with comparatively little investment is mildly terrifying. A force 20, 40 block chance 2-handed bokken (with these current changes) is **MORTIFYINGLY** robust - this is basically a better *captain's sabre* that anyone can make for like 10 wood or whatever.

For extra context, your changes to the hammer have made it stronger than the Wizard's Mjolnir (which has 30 throwforce). Force 36 on every hit is hysterically strong, even accounting for our extra health.

Jousting on the spear is funny and excellent - we should've had that from the start, so good choice there. For observers: jousting by default adds up to 10 damage and 100% knockdown chance (2 seconds) if the person using the weapon is buckled to a moving vehicle and crosses at least 5 tiles between an attack. This means the spear goes up to *33* damage + guaranteed 2s knockdown for one hit if wielded and someone rides a wolf or something.

**Comment by Zergspower** [comment]
i'll maybe peek closer later but - while you're here check their butchering timer because currently it takes 30 seconds to butcher a corpse with forged items for some godamn reason

**Comment by Ephemeralis** [comment]
I also want to stress that a double-edged e-sword is 40 force. An iron reagent hammer is 36 force.

Yeah.

**Comment by Namwty** [comment]
> I also want to stress that a double-edged e-sword is 40 force. An iron reagent hammer is 36 force.

Not wrong, but comparing them to each other isn't exactly ideal. Damage is close, but other factors aren't... like armor penetration, wounding from weapon trait etc, block/reflect. D-sword is way more lethal, even if iron hammer had 40 force, it's not even close.


Arguably, all of their damage should receive small change ( like katana going from 28 to 23/24/25 or something ), Iron hammer could go to like 28/29 at best.


> force 20, 40 block chance 2-handed bokken (with these current changes) is **MORTIFYINGLY** robust - this is basically a better _captain's sabre_ that anyone can make for like 10 wood or whatever.

Why would one handing bokken be more hard hitting than if you were two handing it?

Why 1 handed bokken does 20 dmg, but 2 handed does 10 force? Why is making it other way around, a bad idea?
One handed bokken with 20 dmg and 20% block chance with a telescopic shield/stun baton on other hand, sounds more fair?

Why does fire axe, bone axe, spears follow different logic? Why not make fire axe do more damage one handed, and weaker when wielded? Like am I missing some type of logic here or what, why bokken follows different logic completely?

"A bokken that is capable of blocking attacks when wielding in two hands" description, but you can block with it using one handed.
Also, I never seen or felt it being 40 block chance when wielded, even VV claims that it's 20 block chance in wielded form, so kind of curios on that part.




Other than that, I don't mind these changes that this PR added, tho tweaks on weapon damage would be ideal, I think they are by couple dmg points overtuned ( Especially hammer ). I like that hammer can held one handed, I like dagger and what not actually being tools.


Without this PR, reagent weapons are either **misleading** or worse/on par with stuff that is easier to get.
Old hammer =/< Chainsaw ( easier to make, has sharp edge )
Sword/katana < Cargo blade 
And so on, be it arrows/throwing weapons and what not.

Katana is in fact worst of them all, it literally looks the same as cargo sabre, same force damage, same block chance, same armor penetration. So a lot of people assume that it's basically same a sabre, which isn't even true
Katana

What's best, is that you basically either have to dive into code, be admin or have local to even see this in first place, I wouldn't exactly say this is isn't a frustration to average player. Like would it really hurt to make them at very least same thing, so it doesn't confuse people?

As it currently stands, with or without this PR, post imbue removal forging is borderline just niche content ( cause simply not desirable or interactive enough ). Even if hammer does 36 damage and what not, most people will still choose gun cargo/robotics etc for style and stuff. Hammer <<<< A damn war-crime grenade launcher in the arm.

Yesterday I brought hammer, spear and sword to void raptor mold fight in prison. Only spear was used, rest rotted on the floor, cause most people are not interested in them, even if they hit 36. Guns more fun, wirespool go whip!


TL;DR

Old Reagent weapons sometimes worse than cargo melee and are misleading. This PR is a tad overtuned, but it would make forging a tad bit more desirable and rewarding.

Bokken makes no sense, so why... not fix it?
Why 1 handed bokken with 20 force is ok and balanced, but 2 handed bokken doing 20 force is BROKEN BEYOND ANYTHING.
I couldn't care if 2 handed bokken would have 40 block chance ( which by vv and feeling doesn't seem so ). You would literally be able to get 1 handed bokken with another item in your hand... like riot shield, which has damn 50% block rate ( on top of one handed bokken having 20% block chance ).

Even if dmg is not fine, some weapons counting as tools wouldn't be a bad addition ( like dagger, being a knife ).

**Comment by Ephemeralis** [comment]
I don't think the bokken should be anything higher than f15 wielded and probably like f10 unwielded. It is extremely easy to get (it just requires wood and like 15 seconds of crafting action on a workbench). The block is *huge*. There's an entire suite of scarce chaplain weapons balanced around 10 force and 50 block and we're out here giving people 20 force and 40 block practically for free.

You also have to consider that giving things high force levels means they begin to overcome object hardness as well. For example, the f36 wielded hammer was capable of bashing down blast shutters in 4 hits and airlocks in about 8 hits. We live tested this while it was testmerged. I also saw someone beat *a chef with kitchen-CQC* with the f36 hammer because it instantly mangled their primary arm on hit.

It is perfectly fine to have asymmetrical choices in combat. Melee does not have to compete with guns - they're entirely different balance paradigms. Guns have limited ammunition, they use projectiles that can readily miss, they have a much higher chance of friendly fire and more. Melee has a much shorter range requirement, requires often only a single slot of inventory and has little to no upkeep requirement. Nothing prevents you from teching both simultaneously outside of robustness.

Forging weapons already have their niche as readily attainable baseline weapons. Spears have reach (which are *very* good on this fact alone), katanas have pretty high armor penetration, hammers had decently high base force values, daggers attack super fast, and swords are kind of just middle of the road at everything.

For what it's worth, I've been of the opinion that the shamshir being available in cargo *at all* was a huge mistake.
